pkg/controller/node/node_controller.go

```diff
@@ -519,6 +520,26 @@ func (ctrl *Controller) updateNode(old, cur interface{}) {
 		ctrl.logPoolNode(pool, curNode, "changed from pool %s", oldPool.Name)
 		// Let's also make sure the old pool node counts/status get updated
 		ctrl.enqueueMachineConfigPool(oldPool)
+	} else if err != nil {
+		// getPrimaryPoolForNode may error due to multiple custom pools. In this scenario, let's
+		// queue all of them so that when the node attempts to exit from this error state, the MCP
+		// statuses are updated correctly.
+		klog.Errorf("error fetching old primary pool for node %s, attempting to sync all old pools", oldNode.Name)
+		masterPool, workerPool, customPools, listErr := helpers.ListPools(oldNode, ctrl.mcpLister)
+		if listErr == nil {
+			for _, pool := range customPools {
+				ctrl.enqueueMachineConfigPool(pool)
+			}
+			if masterPool != nil {
+				ctrl.enqueueMachineConfigPool(masterPool)
+			}
+			if workerPool != nil {
+				ctrl.enqueueMachineConfigPool(workerPool)
+			}
+		} else {
+			klog.Errorf("error listing old pools %v for node %s", listErr, oldNode.Name)
```

Comment:
should this return here?

Reply:
no, it's not critical to update the status for old pools, more of a nice to have. The old pools will get queued eventually - we wouldn't want to skip a node update if one of the old pools have an issue here.

This fix just makes the old pool status update immediate vs delayed, for the specific case of when the node previously belonged to two or more pools